File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/model/EraseTool.java
package com.savvasdalkitsis.gameframe.draw.model;

import android.graphics.Color;

public class EraseTool extends AbstractDrawingTool {

    @Override
    public void drawOn(Layer layer, int startColumn, int startRow, int column, int row, int color) {
        int erase = layer.isBackground() ? Color.GRAY : Color.TRANSPARENT;
        layer.getColorGrid().setColor(erase, column, row);
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/feature/draw/model/EraseTool.kt
package com.savvasdalkitsis.gameframe.feature.draw.model

import android.graphics.Color

class EraseTool : AbstractDrawingTool() {

    override fun drawOn(layer: Layer, startColumn: Int, startRow: Int, column: Int, row: Int, color: Int) {
        val erase = if (layer.isBackground) Color.GRAY else Color.TRANSPARENT
        layer.colorGrid.setColor(erase, column, row)
    }
}
